public_html/wp-content/mu-plugins/blocks/assets/src/shared/avatar-size/index.js

```diff
@@ -41,64 +43,82 @@ const sizePresets = [
 	},
 ];
 
-function AvatarSizeControl( {
-	className,
-	label,
-	help,
-	value,
-	initialPosition,
-	onChange,
-	rangeProps,
-} ) {
-	return (
-		<BaseControl
-			className={ classnames( 'wordcamp-components-avatar-size', className ) }
-			label={ label }
-			help={ help }
-		>
-			<div className="wordcamp-components-avatar-size-buttons">
-				<ButtonGroup aria-label={ label }>
-					{ sizePresets.map( ( preset ) => {
-						const { name, shortName, size, slug } = preset;
-						const isCurrent = value === size;
+class AvatarSizeControl extends Component {
+	constructor( props ) {
+		super();
 
-						return (
-							<Button
-								key={ slug }
-								isLarge
-								isPrimary={ isCurrent }
-								aria-label={ name }
-								aria-pressed={ isCurrent }
-								onClick={ () => onChange( Number( size ) ) }
-							>
-								{ shortName || name }
-							</Button>
-						);
-					} ) }
-				</ButtonGroup>
+		this.state = {
+			value    : props.value,
+			onChange : debounce( props.onChange, 10 ) // higher values lead to a noticeable degradation in visual feedback.
 
-				<Button
-					className="wordcamp-components-avatar-size-button-reset"
-					isLarge
-					isDefault
-					onClick={ () => onChange( Number( initialPosition ) ) }
-				>
-					{ __( 'Reset', 'wordcamporg' ) }
-				</Button>
-			</div>
+			// converting from props to state just to debounce feels like maybe it's the wrong way?
+			// maybe related to https://github.com/vasanthk/react-bits/blob/master/anti-patterns/01.props-in-initial-state.md
+			// ^ might also mean that tracking value in state is also wrong?
```

Comment:
Would it work to just put the `debounce` call in the RangeControl? Like:

```
<RangeControl
	...
	onChange={ debounce( this.onChange, 10 ) }
	...
</RangeControl>
```

Tracking the value in state seems ok (we're doing it in `ItemSelect` as well), but this does seem a bit overkill...

Reply:
I tried that first, but then the slider element doesn't update, which would be a really bad UX.